lib/core-data/components.js

```diff
@@ -31,16 +31,19 @@ export function getDefaultData(uri, testStore) {
 }
 
 /**
- * check to see if a string is camelCased
+ * Check if a string contains non-letter, non-number, or non-underscore chars
+ *
  * @param  {string} str
  * @return {boolean}
  */
-function isCamelCase(str) {
-  const camel = _.camelCase(str);
+function isValidKilnDotNotation(str) {
+  // https://regex101.com/r/wapXTM/2 for an example of this regex in action
 
-  // note: we're lowercasing everything because we don't care about 100% perfect camelCase (e.g. imageURL is valid),
-  // we just want to make sure that people aren't using weird characters that cannot be used as properties in dot notation
-  return str.toLowerCase() === camel.toLowerCase() || str.toLowerCase() === `_${camel}`.toLowerCase(); // lodash's camelCase() will remove leading underscores, which we allow in things like _version, _description, and _devDescription
+  // This function is used after "toDotNotation()" in "assertCamelCasedProps()" below
+  // "toDotNotation" actually allows numbers, which are not valid property accessors in dot notation, but which is why we permit numbers here
+
```

Comment:
If we just want to account for this `__` case, what do you think of just adding it to this expression?
```suggestion
return str.toLowerCase() === camel.toLowerCase() || str.toLowerCase() === `_${camel}`.toLowerCase() || str.toLowerCase() === `__${camel}`.toLowerCase(); 
```

Reply:
Also, we'll need to update the corresponding function in claycli: https://github.com/clay/claycli/blob/master/lib/cmd/lint.js#L272

Reply:
Maybe I'm misunderstanding, but that regex should account for an arbitrary number of underscores, no? Is there a scenario I'm missing?

I can put in a PR for ClayCli. There's no place common to both of these repos where we could put this function, is there?

Reply:
Yep. Do you think it's necessary to support more than two underscores? If you just need to add that one case, what do you think of just adding that one `__` case?

Reply:
There were a few things I was thinking about here: 

1. In this particular scenario, we are only supporting two underscores, but what if someone wanted three? Is there a reason we wouldn't want to support an arbitrary number of leading underscores? 

2. 1 sort of gets into the bigger issue, which is that the function was supposedly created to check that a string was camel case, but the description inside of it said that it was there to prevent people from "using weird characters that cannot be used as properties in dot notation". And that's what it did! Even though it was called `isCamelCase`, it didn't actually check that a string was camel cased; it used Lodash's `_.camelCase` as a validator and then compared the lowercased result to the lowercased original string. The purpose of this was to make sure that Schema field names are valid dot notation property accessors (even though, as I note in my comment, numbers aren't valid). 

I can add the double underscore as an exception and it'll work, but this seems like a great opportunity to try to improve a misleading piece of code (and remove an unnecessary use of Lodash). There are many parts of Clay that are very confusing because they have lots of complex and/or procedural steps, and it's always nice when we have the chance to simplify things.